src/adr/0041-james-servers-offering.md

```diff
@@ -0,0 +1,109 @@
+server furthermore is extensible, and ships many extra features by default. It is however hard to operate.
+
+ - **Guice memory** (`server/container/guice/memory-guice`) ships a server based on memory. Easy to boot, 
+without external dependencies, we mostly rely on it for testing purposes.
+
+ - **JPA Guice** (`server/container/guice/jpa-guice`) ships an extensible server based on JPA storage. With
+a default configuration using Derby embedded database, this server have no external dependencies by default.
+However the still large amount of configuration can make new comers confused. It currently satisfies operator 
+looking for self hosting solutions as well as those looking for an extensible server.
+
+ - **JPA SMTP** (`server/container/guice/jpa-smtp`) demonstrate a Mail Transfer Agent, dedicated to mail
+processing, that do not ship the James Mailbox, as well as Mail Delivery protocols like IMAP, JMAP or POP3.
+
+We can notice that this offer is specified in technical terms, and do not directly map to identified user needs.
+
+Also the large amount of servers makes the choice harder to make for newcomers.
+
+## Decision
+
+We should deliver a subset of mail server directly adressing users needs, and brand them according to their use cases
+and not due to their technical details.
+
+In order to get a clearer offer we should first reduce the server cardinality. This can be achieved via:
+
+ - **Spring server** deprecation and removal
+ - Merging Guice servers with their LDAP variations (see [ADR 36 about guice module choosing](0036-against-use-of-conditional-statements-in-guice-modules.md))
+ - **Cassandra-guice** deprecation and removal
+
+We should then rename our servers in order to match the use cases:
+
+ - **Basic server** addresses the self-hosting use case. It will be based on current *jpa-guice* server but with the clear intent to 
+lower adoption barriers. It exposes limited configurations in order not to confuse the user. Configuration and documentation is centered 
+around self hosting use cases.
+
+ - **Advanced server** addresses the customization use cases. It will be based on the current *jpa-guice* server. It differs from the basic server
+as more features will be packaged as part of it, configuration shows via examples advanced usage possibilities, and the documentation is centered 
+around feature extensibility.
+
+Note that transitionning from basic server to the advanced one will be easy as they rely on the same technologies.
+
+ - **Distributed server** will remain unchanged.
+
+ - **Testing server** will rely on current `memory guice` server.
+
+ - **Mail processing server** will rely on current `JPA SMTP guice` server.
+
+Also, we should separate our server application from the technical guice bindings, and group them together for better discoverability.
+
+We should adopt this folder structure:
+
+```
+server/apps
```

Comment:
why server _and_ apps? why not choose only one?

Reply:
I propose `server/distributions` instead. Would it be better?

Reply:
`server/flavor`? :trollface:
more seriously I'm fine with distributions

Reply:
Distribution is how you put things for consumption (like a tar or a zip).
Here we put things together, so `apps` without `server` prefix (because James is a server anyway) would probably better fit.